### PR TITLE
Fix new serve api to retrive endpoints in benchmark

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

Current benchmark is broken on master since `list_endpoints` API is gone in serve's init. 

## Test plan

locally execute both single and multi deployment and ensure they pass.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/serve_tests/workloads/multi_deployment_1k_noop_replica.py [hunk 1]
```diff
@@ -144,7 +144,10 @@ def main(num_replicas: Optional[int], num_deployments: Optional[int],
 
     logger.info("Warming up cluster ....\n")
     rst_ray_refs = []
-    for endpoint in serve.list_endpoints().keys():
+    all_endpoints = list(
+        ray.get(serve_client._controller.get_all_endpoints.remote()).keys())
```
Comment (edoakes) [inline release/serve_tests/workloads/multi_deployment_1k_noop_replica.py:148]: let's not use internal APIs or it'll break. use `serve.list_deployments()`

**Review by edoakes — CHANGES_REQUESTED** [review]

**Comment by simon-mo** [comment]
to catch breakage easier, can you add a dry-run mode that runs through single iteration with small nodes and run add it to CI?

**Comment by simon-mo** [comment]
actually, if you can't find a way to do it in short amount of time, I can do it as a follow up

**Comment by jiaodong** [comment]
> actually, if you can't find a way to do it in short amount of time, I can do it as a follow up

happy to learn how to enable it but probably much faster if you have an example to follow :)

**Comment by simon-mo** [comment]
So Ray tune's example directory is a great example, their script support `--smoke-test` flag and this flag is turned on for running in CI builds (prs and master builds) https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/pbt_example.py

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/BUILD [hunk 2]
```diff
@@ -3,6 +3,7 @@
 py_library(
     name = "serve_lib",
     srcs = glob(["**/*.py"], exclude=["tests/**/*.py"]),
+    visibility = ["//python/ray/serve:__subpackages__", "//release:__pkg__"],
```
Comment (edoakes) [inline python/ray/serve/BUILD:6]: what's this for btw?

**Comment by edoakes** [comment]
@jiaodong ping me when this is ready to merge please.

**Pull request merged**